zrml/global-disputes/src/lib.rs

```diff
@@ -0,0 +1,670 @@
+                    vote_lock_counter,
+                    T::MaxOwners::get(),
+                ))
+                .into())
+            } else {
+                T::Currency::set_lock(
+                    T::GlobalDisputeLockId::get(),
+                    &voter,
+                    lock_needed,
+                    WithdrawReasons::TRANSFER,
+                );
+                Ok(Some(T::WeightInfo::unlock_vote_balance_set(
+                    vote_lock_counter,
+                    T::MaxOwners::get(),
+                ))
+                .into())
+            }
+        }
+    }
+
+    impl<T: Config> Pallet<T> {
+        pub fn reward_account(market_id: &MarketIdOf<T>) -> T::AccountId {
+            T::GlobalDisputesPalletId::get().into_sub_account_truncating(market_id)
+        }
+
+        fn update_winner(market_id: &MarketIdOf<T>, outcome: &OutcomeReport, amount: BalanceOf<T>) {
+            <Winners<T>>::mutate(market_id, |highest: &mut Option<WinnerInfoOf<T>>| {
+                *highest = Some(highest.clone().map_or(
+                    WinnerInfo::new(outcome.clone(), amount),
+                    |prev_winner_info| {
+                        if amount >= prev_winner_info.outcome_info.outcome_sum {
+                            WinnerInfo::new(outcome.clone(), amount)
+                        } else {
+                            prev_winner_info
+                        }
+                    },
+                ));
+            });
+        }
+    }
+
+    impl<T> GlobalDisputesPalletApi<MarketIdOf<T>, AccountIdOf<T>, BalanceOf<T>> for Pallet<T>
+    where
+        T: Config,
+    {
+        fn push_voting_outcome(
+            market_id: &MarketIdOf<T>,
+            outcome: OutcomeReport,
+            owner: &T::AccountId,
+            initial_vote_balance: BalanceOf<T>,
+        ) -> DispatchResult {
+            match <Winners<T>>::get(market_id) {
+                Some(winner_info) if winner_info.is_finished => {
+                    return Err(Error::<T>::GlobalDisputeAlreadyFinished.into());
+                }
+                _ => (),
+            }
+            match <Outcomes<T>>::get(market_id, &outcome) {
+                Some(mut outcome_info) => {
+                    let outcome_sum = outcome_info.outcome_sum.saturating_add(initial_vote_balance);
+                    Self::update_winner(market_id, &outcome, outcome_sum);
+                    outcome_info.outcome_sum = outcome_sum;
+                    if outcome_info.owners.try_push(owner.clone()).is_ok() {
+                        <Outcomes<T>>::insert(market_id, outcome, outcome_info);
+                    } else {
+                        log::warn!(
+                            "Global Disputes: The voting outcome was not added.  This happens \
+                             because there are too many voting outcome owners (length is {:?}).",
+                            &outcome_info.owners.len()
+                        );
```

Comment:
> Good question. The thought was, that global disputes is the resolution mechanism of last resort. So I wanted to make sure, that there are as few error cases as possible to actually finally get a decision over a dispute.

I fear the case, that the global dispute doesn't start because of exceeding the owner length. I would rather start the global dispute and ignore the case, that one or two owners were not added to the owner list. 
But I mean this would be a mistake of the developer (controls `MaxDisputes`).

I am open minded about this. So let me fix that. 

db79f1b72a568d37fc8cabb85adbadbd96f87bc2

Reply:
I see, thanks for explaining and fixing! I think it's better to error. The real problem here is that it's possible to misconfigure the pallets.